### ISSUE TITLE
support for ASOF joins

### ISSUE THREAD

**Issue by era127** [PRQL/prql#3390]
### What's up?

I am starting to look at PRQL and I was wondering if there is the possibility to support ASOF joins such as in [duckdb](https://duckdb.org/docs/guides/sql_features/asof_join.html)   I understand this type of support is not in standard sql but It would help to have it translated.

**Comment by max-sixty** [comment]
Yes I think this makes sense, and would be quite easy — maybe as easy as adding it to https://github.com/PRQL/prql/blob/04b5174cba01bcc79837e69f0697486210d3d6dd/crates/prql-compiler/src/ir/pl/extra/expr.rs#L112-L118 and catching any places where that's matched against...

**Comment by xueyumusic** [comment]
It looks depending on sqlparser crate [here](https://github.com/PRQL/prql/blob/main/prqlc/prqlc/src/sql/gen_query.rs#L344). So does it mean need to add support in sqlparser crate first?

**Comment by max-sixty** [comment]
> It looks depending on sqlparser crate [here](https://github.com/PRQL/prql/blob/main/prqlc/prqlc/src/sql/gen_query.rs#L344). So does it mean need to add support in sqlparser crate first?

Coming back to this issue — this is correct... I think that would likely be a welcome contribution for them, and not that much code...

**Comment by max-sixty** [comment]
FYI this is now released upstream! So could be a reasonable contribution to PRQL

https://github.com/sqlparser-rs/sqlparser-rs/pull/1288
